internal/advtools/fastcombat.lua

```diff
@@ -0,0 +1,91 @@
+
+-- Overlay
+AdvCombatOverlay = defclass(AdvCombatOverlay, overlay.OverlayWidget)
+AdvCombatOverlay.ATTRS{
+    desc='Faster combat!',
+    default_enabled=true,
+    viewscreens='dungeonmode',
+    frame={w=0, h=0},
+}
+local skip_combat = false
+function AdvCombatOverlay:render()
+    local adv = df.global.adventure
+    if adv.player_control_state == df.adventurest.T_player_control_state.TAKING_INPUT then
+        skip_combat = false
+        return
+    end
+    if skip_combat then
+        -- Instantly process the projectile travelling
+        adv.projsubloop_visible_projectile = false
+        -- Skip the combat swing animations
+        adv.game_loop_animation_timer_start = adv.game_loop_animation_timer_start + 1000
+    end
+end
+
+
+COMBAT_MOVE_KEYS={
+    _MOUSE_L = true,
+    SELECT = true,
+    A_MOVE_N = true,
+    A_MOVE_S = true,
+    A_MOVE_E = true,
+    A_MOVE_W = true,
+    A_MOVE_NW = true,
+    A_MOVE_NE = true,
+    A_MOVE_SW = true,
+    A_MOVE_SE = true,
+    A_MOVE_SAME_SQUARE = true,
+    A_ATTACK = true,
+    A_COMBAT_ATTACK = true,
+}
+
+function AdvCombatOverlay:onInput(keys)
+    if AdvCombatOverlay.super.onInput(self, keys) then
+        return true
+    end
+    local adv = df.global.adventure
+    for code,_ in pairs(keys) do
+        if COMBAT_MOVE_KEYS[code] then
+            if adv.player_control_state ~= df.adventurest.T_player_control_state.TAKING_INPUT then
+                -- Instantly speed up the combat
+                skip_combat = true
+            elseif adv.player_control_state == df.adventurest.T_player_control_state.TAKING_INPUT then
+                if COMBAT_MOVE_KEYS[code] and df.global.world.status.temp_flag.adv_showing_announcements then
+                    -- We're using mouse to skip, more unique behavior to mouse clicking is handled here
+                    if keys._MOUSE_L then
+                        x,y = dfhack.screen.getMousePos()
+                        local screen_width, _ = dfhack.screen.getWindowSize()
+                        -- Calculate if we're clicking within the vanilla adv announcements box
+                        adv_announce_width = 112
+                        adv_announce_x1 = screen_width/2 - adv_announce_width/2 - 1
+                        adv_announce_x2 = screen_width/2 + adv_announce_width/2 - 1
+                        -- The Y values for this box don't change
+                        adv_announce_y1, adv_announce_y2 = 6, 20
+                        -- Check if we're clicking within the bounds of the adv announcements box that is being shown right now
+                        if y >= adv_announce_y1 and y <= adv_announce_y2 and x >= adv_announce_x1 and x <= adv_announce_x2 then
+                            -- Don't do anything on our end, the player is clicking within the adv announcements box.
+                            -- We don't want to overtake vanilla behavior in this ui element.
+                            return
+                        end
```

Comment:
please use a widgets.Panel for this. Example: https://github.com/DFHack/scripts/blob/master/internal/exportlegends/asyncexport.lua#L65

Reply:
I don't get it

Reply:
I could not figure it out. Any self:addviews I added refused to show up, there were no errors in console and I've spent an hour just trying and failing to get a label displayed on screen.